File: tests/fixtures/objc/embedded_block.m
#import <Foundation/Foundation.h>

@implementation TemplateRenderer

- (NSString *)renderPage:(NSString *)title {
    NSString *html = @"<html>\n"
        "<head>\n"
        "  <meta charset='utf-8'>\n"
        "  <title>Page</title>\n"
        "  <style>\n"
        "    body { margin: 0; padding: 20px; }\n"
        "    .container { max-width: 960px; }\n"
        "    .header { background: #333; color: #fff; }\n"
        "    .nav { display: flex; }\n"
        "    .content { padding: 20px; }\n"
        "    .footer { border-top: 1px solid #ccc; }\n"
        "    .sidebar { width: 200px; }\n"
        "  </style>\n"
        "</head>\n"
        "<body>\n"
        "  <div class='container'></div>\n"
        "</body>\n"
        "</html>";
    return html;
}

@end

File: tests/fixtures/swift/embedded_block.swift
func buildTemplate() -> String {
    let template = """
    <html>
    <head>
        <title>Page</title>
        <meta charset="utf-8">
        <link rel="stylesheet" href="style.css">
        <script src="app.js"></script>
    </head>
    <body>
        <header>
            <h1>Title</h1>
            <nav>Home | About | Contact</nav>
        </header>
        <main>
            <section>Content goes here</section>
        </main>
        <footer>
            <p>Copyright 2024</p>
        </footer>
    </body>
    </html>
    """
    return template
}
